native/react-native/common-components/modals/confirm-modal.js

```diff
@@ -0,0 +1,185 @@
+    configureConfirmModal: (modalStateObject) => dispatch(configureConfirmModal(modalStateObject))
+});
+
+//We use this to reset the modal when we're done with it.
+const defaultModalState = {
+    isConfirmModalVisible: false,
+    titleMessage: "",
+    bodyMessage: "",
+    confirmFunction: null,
+    confirmMessage: "",
+    cancelMessage: ""
+};
+
+const styles = StyleSheet.create({
+    bodyMessageText: {
+        color: "#000",
+        fontFamily: "Oswald-Regular",
+        paddingHorizontal: 12,
+        textAlign: "center",
+    },
+    button: {
+        alignItems: "center",
+        borderColor: "#000",
+        borderRadius: 4,
+        borderWidth: 2,
+        justifyContent: "center",
+        paddingHorizontal: 25,
+        paddingVertical: 10,
+        width: "40%"
+    },
+    buttonText: {
+        fontFamily: "Lato-Bold"
+    },
+    modalBody: {
+        alignItems: "center",
+        flex: 0.75,
+        justifyContent: "center",
+        padding: 20
+    },
+    modalButtons: {
+        alignSelf: "center",
+        flexDirection: "row",
+        justifyContent: "space-around",
+        marginBottom: 10,
+        width: "100%",
+    },
+    modalHeader: {
+        alignItems: "center",
+        backgroundColor: "#333",
+        borderTopLeftRadius: 5,
+        borderTopRightRadius: 5,
+        justifyContent: "center"
+    },
+    modalHeaderText: {
+        color: "#FFF",
+        fontFamily: "Lato-Bold",
+        fontSize: 18,
+        fontWeight: "bold",
+        padding: 12,
+        textAlign: "center"
+    },
+    modalProper: {
+        alignItems: "center",
+        backgroundColor: "rgba(0,0,0,0.7)",
+        flex: 1,
+        flexDirection: "column",
+        justifyContent: "center"
+    },
+    modalWrap: {
+        backgroundColor: "#FFF",
+        borderRadius: 5,
+        height: "25%",
+        justifyContent: "flex-start",
+        width: "85%",
+    }
+});
+
+/**
+ * This component serves as a global CONFIRM/DENY modal for the application. It is styled to match a basic black/white theme
+ * and provides placeholder text for all fields in case we just want to ask the user if they're really sure. If we have
+ * custom messaging for the user, we can pass it in as props. Since this modal is global we hook into Redux to provide the props.
+ * 
+ * @props:
+ *  - configureConfirmModal: Redux reducer function to configure the modal.
+ *  -  modalState: Object with the following properties to configure the modal:
+ *    - isConfirmModalVisible: Boolean that controls whether modal is visible.
+ *    - titleMessage: Message that is displayed in the header.
+ *    - bodyMessage: Message that is displayed in the main body.
+ *    - confirmFunction: Function that is invoked when user confirms that they want to perform the task
+ *    - confirmMessage: Text displayed in the confirm button
+ *    - cancelMessage: Text displayed in the cancel button
+ */
+
+class ConfirmModal extends Component {
```

Comment:
Normally, when I create a component that does not have any State or needs any functional hooks, I just use a straight up function component. 

While it doesn't change the functionality, when I'm reading the code it does help me understand that this component is just a pure view.

Reply:
🤦‍♂ _That_ is a good point!!

Reply:
Done!